sigstore/_internal/trust.py

```diff
@@ -323,28 +333,39 @@ def __init__(self, inner: _SigningConfig):
         @api private
         """
         self._inner = inner
-        self._verify()
-
-    def _verify(self) -> None:
-        """
-        Performs various feats of heroism to ensure that the signing config
-        is well-formed.
-        """
 
         # must have a recognized media type.
         try:
             SigningConfig.SigningConfigType(self._inner.media_type)
         except ValueError:
             raise Error(f"unsupported signing config format: {self._inner.media_type}")
 
-        # currently not supporting other select modes
-        # TODO: Support other modes ensuring tsa_urls() and tlog_urls() work
-        if self._inner.rekor_tlog_config.selector != ServiceSelector.ANY:
-            raise Error(
-                f"unsupported tlog selector {self._inner.rekor_tlog_config.selector}"
-            )
-        if self._inner.tsa_config.selector != ServiceSelector.ANY:
-            raise Error(f"unsupported TSA selector {self._inner.tsa_config.selector}")
+        # Create lists of service protos that are valid & supported by this client
+        # Limit the TSA and tlog lists using the service selector config
+        self._tlogs = self._get_valid_services(
+            self._inner.rekor_tlog_urls, REKOR_VERSIONS
+        )
+        if not self._tlogs:
+            raise Error("No valid Rekor transparency log found in signing config")
+        if self._inner.rekor_tlog_config.selector == ServiceSelector.EXACT:
+            if len(self._tlogs) < self._inner.rekor_tlog_config.count:
+                raise Error(
+                    "Not enough Rekor transparency logs found in signing config"
+                )
+            self._tlogs = self._tlogs[: self._inner.rekor_tlog_config.count]
+        elif self._inner.rekor_tlog_config.selector == ServiceSelector.ANY:
+            self._tlogs = self._tlogs[:1]
+
+        self._tsas = self._get_valid_services(self._inner.tsa_urls, TSA_VERSIONS)
+        if self._inner.tsa_config.selector == ServiceSelector.EXACT:
+            self._tsas = self._tsas[: self._inner.tsa_config.count]
+        elif self._inner.tsa_config.selector == ServiceSelector.ANY:
+            self._tsas = self._tsas[:1]
```

Comment:
Consider abstracting this into a separate function, like

```python
def _eveluluate_selector(services, selector)
```

Reply:
I can handle it in `_get_valid_services()` if I provide the selectorconfig maybe I'll try that